diff --git a/entity_legal.entity_controller.inc b/entity_legal.entity_controller.inc
index b5317ea..529d963 100644
--- a/entity_legal.entity_controller.inc
+++ b/entity_legal.entity_controller.inc
@@ -11,7 +11,7 @@ class EntityLegalDocumentController extends EntityAPIControllerExportable {
   /**
    * {@inheritdoc}
    */
-  public function save($entity) {
+  public function save($entity, DatabaseTransaction $transaction = NULL) {
     // When creating a new legal document, add the document text to the bundle.
     if (!empty($entity->is_new)) {
       $instance = array(
@@ -38,7 +38,7 @@ class EntityLegalDocumentController extends EntityAPIControllerExportable {
       field_create_instance($instance);
     }
 
-    $success = parent::save($entity);
+    $success = parent::save($entity, $transaction);
 
     // Flush the entity info cache to allow the new bundle to be registered.
     entity_info_cache_clear();
