diff -bur a/googleanalytics.admin.inc b/google_analytics/googleanalytics.admin.inc
--- a/googleanalytics.admin.inc	2018-07-13 09:18:13.000000000 +0300
+++ b/googleanalytics.admin.inc	2020-07-14 17:59:37.242869049 +0300
@@ -193,7 +193,34 @@
     '#default_value' => variable_get('googleanalytics_trackuserid', 0),
     '#description' => t('User ID enables the analysis of groups of sessions, across devices, using a unique, persistent, and non-personally identifiable ID string representing a user. <a href="@url">Learn more about the benefits of using User ID</a>.', array('@url' => 'https://support.google.com/analytics/answer/3123663')),
   );
-
+  // GA User-ID Format.
+  $option_desc = array(
+    t('<strong>"Hash"</strong> is default and most safe option.'),
+    t('<strong>"Plain Drupal UID"</strong> could be used for debugging '
+      . 'or when PII (personally identifiable information) can\'t be '
+      . 'restored by Drupal UID by others (including Google).'),
+    t('<strong>"Drupal UID + Hash"</strong> is a compromize between '
+      . 'security and usability.'),
+  );
+  $form['tracking']['user_vis_settings']['googleanalytics_userid_format'] = array(
+    '#type'  => 'select',
+    '#title' => t('Format of User-ID'),
+    '#default_value' => variable_get('googleanalytics_userid_format', 'hash'),
+    '#options' => array(
+      'hash' => t('Hash (Example: "2CUuFCvrQxgCAdcbx46VdQ8KMD56-l8EXq1kHxBB06A")'),
+      'drupal_uid' => t('Plain Drupal UID (Example: "123")'),
+      'combined' => t('Drupal UID + Hash (Example: "123-2CUuFCvrQxgCAdcbx46VdQ8KMD56-l8EXq1kHxBB06A")'),
+    ),
+    '#description' => theme('item_list', array('items' => $option_desc))
+      . t('<strong>Warning:</strong> "In Google Analytics we\'re not allowed to '
+      . 'track personally identifiable information (PII) such as usernames that '
+      . 'would be identifiable by a 3rd party such as Google."<br />'
+      . 'See <a href="@terms_of_service_link">Section 7 of Terms of service</a>',
+      array(
+        '@terms_of_service_link' => 'https://marketingplatform.google.com/about/analytics/terms/us/',
+      )
+    )
+  );
   // Link specific configurations.
   $form['tracking']['linktracking'] = array(
     '#type' => 'fieldset',
diff -bur a/googleanalytics.install b/googleanalytics.install
--- a/googleanalytics.install	2018-07-13 09:18:13.000000000 +0300
+++ b/googleanalytics.install	2020-07-14 17:59:37.242869049 +0300
@@ -34,6 +34,7 @@
   variable_del('googleanalytics_tracklinkid');
   variable_del('googleanalytics_trackurlfragments');
   variable_del('googleanalytics_trackuserid');
+  variable_del('googleanalytics_userid_format');
   variable_del('googleanalytics_trackmailto');
   variable_del('googleanalytics_trackmessages');
   variable_del('googleanalytics_trackoutbound');
diff -bur a/googleanalytics.module b/googleanalytics.module
--- a/googleanalytics.module	2018-07-13 09:18:13.000000000 +0300
+++ b/googleanalytics.module	2020-07-14 17:59:37.242869049 +0300
@@ -313,7 +313,7 @@

     // Track logged in users across all devices.
     if (variable_get('googleanalytics_trackuserid', 0) && user_is_logged_in()) {
-      $create_only_fields['userId'] = google_analytics_user_id_hash($user->uid);
+      $create_only_fields['userId'] = googleanalytics_get_user_id();
     }

     // Create a tracker.
@@ -709,3 +709,31 @@

   return TRUE;
 }
+
+/**
+ * Get formatted value of GA User-ID.
+ *
+ * @param int $uid
+ *   Drupal UID. Optional.
+ *
+ * @return string
+ *   Returns empty string when Tracking UserID is disabled or   anonymous user.
+ *   Returns formatted value according to specified format.
+ */
+function googleanalytics_get_user_id($uid = '') {
+  // Do not process anonymous users because they all have the same
+  // hash and Drupal UID.
+  if (empty($uid)) {
+    return '';
+  }
+  // Hashed User-ID is a default.
+  $userId = google_analytics_user_id_hash($uid);
+  $format = variable_get('googleanalytics_userid_format', 'hash');
+  if ($format == 'drupal_uid') {
+    $userId = $uid;
+  }
+  elseif ($format == 'combined') {
+    $userId = $uid . '-' . $userId;
+  }
+  return $userId;
+}
