diff --git a/restrict_by_ip.module b/restrict_by_ip.module
index 8c489de..f065fc9 100644
--- a/restrict_by_ip.module
+++ b/restrict_by_ip.module
@@ -192,7 +192,7 @@ function restrict_by_ip_general_settings() {
     '#type' => 'textfield',
     '#title' => t('Header to check'),
     '#description' => t("This is the HTTP request header that contains the client IP Address.  It is sometimes re-written by reverse proxies and Content Distribution Networks.  If it is left blank it will be default to REMOTE_ADDR.  In most cases you can leave this blank."),
-    '#default_value' => variable_get('restrict_by_ip_header', 'REMOTE_ADDR'),
+    '#default_value' => variable_get('restrict_by_ip_header'),
   );
   return system_settings_form($form);
 }
@@ -531,12 +531,13 @@ function _restrict_by_ip_login(&$user) {
  * Returns the IP address of the user, taking into account header configuration.
  */
 function _restrict_by_ip_get_ip() {
-  $header = variable_get('restrict_by_ip_header', 'REMOTE_ADDR');
-  $ip_address = '';
-
-  if (!empty($_SERVER[$header])) {
+  $header = variable_get('restrict_by_ip_header');
+  if ($header) {
     $ip_address = $_SERVER[$header];
   }
+  else {
+    $ip_address = ip_address();
+  }
 
   return $ip_address;
 }
