diff --git a/plugins/contexts/user.inc b/plugins/contexts/user.inc
index bbdbd1a..9cd7308 100644
--- a/plugins/contexts/user.inc
+++ b/plugins/contexts/user.inc
@@ -41,10 +41,13 @@ function ctools_context_create_user($empty, $data = NULL, $conf = FALSE) {
   if ($conf) {
     if ($data['type'] == 'current') {
       global $user;
-      $data = user_load($user->uid);
       if (user_is_logged_in()) {
+        $data = user_load($user->uid);
         $data->logged_in_user = TRUE;
       }
+      else {
+        $data = drupal_anonymous_user();
+      }
     }
     else {
       $data = user_load($data['uid']);
