diff --git a/search_api.module b/search_api.module
index c1cf4e4b..ee296492 100644
--- a/search_api.module
+++ b/search_api.module
@@ -939,6 +939,24 @@ function search_api_entity_delete($entity, $type) {
   }
 }
 
+/**
+ * Implements hook_field_reference_delete_update_parent_entities().
+ *
+ * Marks as "changed" any entities that referenced a just-deleted entity.
+ *
+ * @param array $entities
+ *   The deleted entities.
+ */
+function search_api_field_reference_delete_update_parent_entities($entities) {
+  if (is_array($entities)) {
+    foreach ($entities as $entity) {
+      search_api_track_item_change($entity->entity_type, array($entity->entity_id));
+      $combined_id = $entity->entity_type . '/' . $entity->entity_id;
+      search_api_track_item_change('multiple', array($combined_id));
+    }
+  }
+}
+
 /**
  * Implements hook_node_access_records_alter().
  *
